streamz/tests/test_kafka.py

```diff
@@ -149,16 +159,15 @@ def test_from_kafka_thread():
     ARGS = {'bootstrap.servers': 'localhost:9092',
             'group.id': 'streamz-test%i' % j}
     with kafka_service() as kafka:
+        kafka, TOPIC = kafka
         stream = Stream.from_kafka([TOPIC], ARGS)
         out = stream.sink_to_list()
         stream.start()
         for i in range(10):
             kafka.produce(TOPIC, b'value-%d' % i)
         kafka.flush()
         # it takes some time for messages to come back out of kafka
-        startlen = len([o for o in out if o])
-        wait_for(lambda: len([o for o in out if o]) == startlen + 10, 10,
-                 period=0.1)
+        wait_for(lambda: len(out) == 10, 10, period=0.1)
```

Comment:
If you're willing, we might want to use `await_for` here instead if this is an async test.

Reply:
```suggestion
        yield await_for(lambda: len(out) == 10, 10, period=0.1)
```